model/src/main/java/com/epam/esm/model/dao/GiftCertificateDao.java

```diff
@@ -0,0 +1,28 @@
+package com.epam.esm.model.dao;
+
+import com.epam.esm.model.dao.entity.GiftCertificate;
+import com.epam.esm.model.dao.entity.SortType;
+import com.epam.esm.model.dao.exception.DaoException;
+import com.epam.esm.model.service.dto.CertificateDTO;
+
+import java.util.List;
+import java.util.Map;
+
+public interface GiftCertificateDao {
+
+    GiftCertificate create(GiftCertificate certificate) throws DaoException;
+
+    GiftCertificate read(Long id) throws DaoException;
```

Comment:
It is better to rerurn Optional<GiftCertificate>, in the dao level, when you find a single entity (by id, on example).